src/Interactive/EditorFeatures/Core/Extensibility/Interactive/InteractiveEvaluator.cs

```diff
@@ -260,50 +260,58 @@ private Dispatcher Dispatcher
 
         private void SubmissionBufferAdded(object sender, SubmissionBufferAddedEventArgs args)
         {
-            AddSubmission(_currentWindow.TextView, args.NewBuffer);
+            AddSubmission(_currentWindow.TextView, args.NewBuffer, this.LanguageName);
         }
 
         // The REPL window might change content type to host command content type (when a host command is typed at the beginning of the buffer).
         private void LanguageBufferContentTypeChanged(object sender, ContentTypeChangedEventArgs e)
         {
             var buffer = e.Before.TextBuffer;
-            var afterIsCSharp = e.AfterContentType.IsOfType(ContentTypeNames.CSharpContentType);
-            var beforeIsCSharp = e.BeforeContentType.IsOfType(ContentTypeNames.CSharpContentType);
 
-            if (afterIsCSharp == beforeIsCSharp)
+            var afterIsLanguage = e.AfterContentType.IsOfType(this.ContentType.TypeName);
+            var afterIsInteractiveCommand = e.AfterContentType.IsOfType(PredefinedInteractiveCommandsContentTypes.InteractiveCommandContentTypeName);
+            var beforeIsLanguage = e.BeforeContentType.IsOfType(this.ContentType.TypeName);
+            var beforeIsInteractiveCommand = e.BeforeContentType.IsOfType(PredefinedInteractiveCommandsContentTypes.InteractiveCommandContentTypeName);
+
+            if (afterIsLanguage == beforeIsLanguage ||
```

Comment:
Don't we essentially want to bail out of this method if the content type is not changing from or to InteractiveCommand?

 Should this condition be 
if (!(beforeIsInteractiveCommand != afterIsInteractiveCommand))
{
      return;
}